src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -454,7 +455,10 @@ private uint ReportTcpSNIError(string errorMessage)
 
         private uint ReportErrorAndReleasePacket(SNIPacket packet, string errorMessage)
         {
-            packet.Release();
+            if (packet != null)
+            {
+                packet.Release();                
```

Comment:
In that case, +1 to doing *that* :) I'm curious what the blocker, if any, is...is there one?